### ISSUE TITLE
Incorrect (inverse) patches when model action in onAttachedToRootStore

### ISSUE THREAD

**Issue by sisp** [xaviergonz/mobx-keystone#85]
When a model action is performed in `onAttachedToRootStore`, the order in which `onPatches` emits patches is unexpected (to me) and the order in which it emits inverse patches is wrong, i.e. it is impossible to revert the changes by applying inverse patches (in reverse order - see #84).

I believe that model actions performed in `onAttachedToRootStore` should trigger a patch _after_, e.g., the patch for adding a model instance to a tree (with a root store) is triggered. This issue may be related to #76.

This is a test case that demonstrates the problem:

```ts
test("patches with action in onAttachedToRootStore", () => {
  @model("test/patchesWithActionInOnAttachedToRootStore/M")
  class M extends Model({ value: prop<number>(0) }) {
    onAttachedToRootStore() {
      this.setValue(1)
    }

    @modelAction
    setValue(value: number) {
      this.value = value
    }
  }

  @model("test/patchesWithActionInOnAttachedToRootStore/R")
  class R extends Model({ ms: prop<M[]>(() => []) }) {
    @modelAction
    addM(m: M) {
      this.ms.push(m)
    }
  }

  const r = new R({})
  autoDispose(() => unregisterRootStore(r))
  registerRootStore(r)

  const sn = getSnapshot(r)

  const rPatches: Patch[][] = []
  const rInvPatches: Patch[][] = []
  autoDispose(
    onPatches(r, (ptchs, iptchs) => {
      rPatches.push(ptchs)
      rInvPatches.push(iptchs)
    })
  )
  expect(rPatches).toMatchInlineSnapshot(`Array []`)
  expect(rInvPatches).toMatchInlineSnapshot(`Array []`)

  r.addM(new M({}))

  runUnprotected(() => {
    rInvPatches
      .slice()
      .reverse()
      .forEach(invpatches => applyPatches(r, invpatches, true))
  })
  expect(getSnapshot(r)).toStrictEqual(sn)
})
```

**Comment by xaviergonz** [comment]
Does the PR you made fix this issue?

**Comment by sisp** [comment]
Just to clarify, #84 doesn't solve this problem. I was able to create a test case that demonstrates the problem, but I'm not sure what needs to be done to fix it.

**Comment by sisp** [comment]
As #84 has been merged, I've updated the test case to use the new `reverse` flag of `applyPatches`.

**Comment by xaviergonz** [comment]
should be fixed in 0.33.0, thanks!

**Comment by sisp** [comment]
Perfect, thanks!
